source/vir/src/context.rs

```diff
@@ -70,6 +71,11 @@ pub struct Ctx {
     pub(crate) debug: bool,
     pub fun: Option<FunctionCtx>,
     pub global: GlobalCtx,
+    // In the very unlikely case where we get sha512 collisions
+    // we use this to panic rather than introduce unsoundness.
+    // Of course it can be argued that accounting for sha512 collisions
+    // is overkill, perhaps this should be revisited.
+    pub(crate) string_hashes: RefCell<HashMap<BigUint, Arc<String>>>,
```

Comment:
Is this unnecessary? a sha512 collision is extremely unlikely and even feels a bit silly to account for